backend/functions/index.js

```diff
@@ -52,3 +53,29 @@ exports.updateMostRecentMessage = functions.database.ref('/messages/{messageId}'
 
 		return ref.child(message.conversation).child("lastMessage").set(message);
 	});
+
+/**
+ * Sets a unique username for the newly created user
+ */
+exports.setUniqueUsername = functions.auth.user().onCreate(event => {
+	var ref = admin.database().ref("users");
+
+	const id = event.data.uid;
+	const email = event.data.email;
+	// string before '@' symbol with trailing digits removed
+	const potentialUsername = email.substring(0, email.indexOf('@')).replace(/\d+$/, '');
+
+	ref.once("value", function(snapshot) {
+		var matchedUsernames = 0;
+		snapshot.forEach(function(data) {
+			const user = data.val();
+			if (user.username !== undefined && user.username.startsWith(potentialUsername)) {
```

Comment:
`bob1@yahoo.com` will be assigned the username 'bob' because the regex removes the trailing digits. If someone signs up with 'bob@gmail.com' then, they will be assigned the username 'bob1'.

`bob1@yahoo.com` won't be assigned 'bob1' unless there's another 'bob' in the database. And if there's another 'bob' in the database, then `bob@gmail.com` will be assigned 'bob2'.